### DISCUSSION TITLE
Inactivity Settings

### DISCUSSION THREAD

**Discussion by jbk68** [cortexrd/Knack-Toolkit-Library#190] [General]
Just wondering if there currently is way using KTL that allows me to set different Inactivity timing settings for certain user roles.

Would be nice if for "Admin - Level_1"  the inactivity setting could set the to 90 minutes.  For user roles "Admin - Level_2", theirs would be set to 60 minutes and for user roles "Customers" they would be set to 30 minutes.

Knack allows you to change inactivity timing from "OFF" up "60 Minutes" but one settings applies to ALL user roles.

Thank you... JON

**Comment by cortexrd** [comment]
Hi Jon,

I don't have a keyword for this yet, but it could be a good idea.

In the meantime, add these few lines to your Javascript code inside the **//KTL Setup - BEGIN... END** section.
Note that you'll also have to disable the auto log-out feature in the Builder.

```
const roles = Knack.getUserRoleNames();
var idleWatchDogDelay = 120; //Value in minutes. 2 hours by default.
if (roles.includes('Admin'))
    idleWatchDogDelay = 60;
else if (roles.includes('Supervisor'))
    idleWatchDogDelay = 15;
else if (roles.includes('Client') || roles.includes('Prospect') || roles.includes('Manager')) //All these have the same value.
    idleWatchDogDelay = 5;
else if (roles.includes('Developer'))
    idleWatchDogDelay = 0; //Disabled for Developers

ktl.scenes.setCfg({
    idleWatchDogDelay: idleWatchDogDelay * 60000,
    //idleWatchDogTimeout: idleWatchDogTimeout, //This is the function you'll need to implement, depending on your choice below.
})
```

The next question is:  what do you want to the app to do after the inactivity expires?
1. Logout?
2. Refresh the page?
3. Redirect to another page?
4. Other

This is where the function idleWatchDogTimeout() will come into play.

If you want to stick to the default action, you can use this function:

```
//This is called after there's been no keypress or mouse click for a certain period of time.
//By default it does nothing.
function idleWatchDogTimeout() {
    ktl.account.logout(); //As an example, here we force a logout.
}
```

Just add it inside the section **//KTL callbacks to your App - BEGIN... END**

**As a bonus**, you won't ever be kicked out of the Builder anymore when you're working in the App for a long time.  That is soooo annoying!

Enjoy,
Norm

**Reply by jbk68** [reply]
Hey Norm, appreciate the solution.  I'll give it a try.

Thanks... JON
